src/components/Data/Indexes/Page.vue

```diff
@@ -223,69 +229,80 @@ export default {
     }
   },
   mounted() {
-    this.updateFilteredIndexes(this.tableItems)
+    this.updateFilteredIndexes(this.indexes)
   },
   computed: {
     ...mapGetters('auth', ['canSearchIndex', 'canCreateIndex']),
-    loading() {
-      return this.$store.direct.state.index.loadingIndexes
-    },
+    ...mapGetters('index', ['indexes', 'loadingIndexes']),
     bulkDeleteEnabled() {
       return this.selectedIndexes.length > 0
     },
-    indexes() {
-      return this.$store.state.index.indexesAndCollections
-    },
-    tableItems() {
-      return Object.keys(this.indexes).map(i => ({
-        indexName: i,
-        collectionCount:
-          this.indexes[i].realtime.length + this.indexes[i].stored.length
-      }))
-    },
-    orderedFilteredIndices() {
-      return this.$store.state.index.indexes
-        .filter(indexName => indexName.indexOf(this.filter) !== -1)
-        .sort()
+    allChecked() {
+      if (!this.selectedIndexes || !this.filteredIndexes) {
+        return false
+      }
+      return this.selectedIndexes.length === this.filteredIndexes.length
     }
   },
   methods: {
     openCreateModal() {
       this.$bvModal.show(this.createIndexModalId)
     },
-    onModalClose() {
-      this.selectedIndexes.shift()
-      this.deleteIndexes()
+    async onDeleteModalClose() {
+      this.selectedIndexes = this.selectedIndexes.slice(1)
+
+      if (this.bulkDeleteEnabled) {
+        this.deleteIndexes()
+      } else {
+        await this.listIndexes()
+      }
     },
-    deleteIndexes() {
-      if (this.selectedIndexes.length > 0) {
-        this.openDeleteModal(this.selectedIndexes[0])
+    async onCreateModalClose() {
+      await this.listIndexes()
+    },
+    async listIndexes() {
+      try {
+        await this.$store.direct.dispatch.index.listIndexes()
+      } catch (err) {
+        this.$log.error(err)
+        this.$bvToast.toast(
+          'The complete error has been printed to the console.',
+          {
+            title: err.message,
+            variant: 'danger',
+            toaster: 'b-toaster-bottom-right',
+            appendToast: true
+          }
+        )
       }
     },
-    allChecked() {
-      if (!this.selectedIndexes || !this.filteredIndexes) {
-        return false
+    deleteIndexes() {
+      if (this.bulkDeleteEnabled) {
+        this.openDeleteModal(this.selectedIndexes[0])
```

Comment:
Why showing the modal only if `bulkDeleteEnabled`? The usual way I do deletion with optional bulk is
* use a `itemsToDelete` array in `data()`
* the modal takes `itemToDelete` as a prop
* if `itemsToDelete.length === 1`, the modal asks the name of the item to confirm deletion
* else, the modal just shows a warning message before deleting

I don't really see the use for a boolean telling whether we're bulk deleting or not.